test/gurka/test_fixed_speed.cc

```diff
@@ -0,0 +1,78 @@
+#include "baldr/graphconstants.h"
+#include "gurka.h"
+#include "test.h"
+#include <boost/format.hpp>
+#include <gtest/gtest.h>
+
+using namespace valhalla;
+
+class FixedSpeedTest : public ::testing::Test {
+protected:
+  static gurka::map speed_map;
+
+  static void SetUpTestSuite() {
+    constexpr double gridsize = 10;
+
+    const std::string ascii_map = R"(
+      B------C
+      |      |
+      |      |
+      A      D
+      |      |
+      E------F
+    )";
+
+    const gurka::ways ways = {{"ABCD", {{"highway", "residential"}, {"maxspeed", "45"}}},
+                              {"AEFD", {{"highway", "residential"}, {"maxspeed", "20"}}}};
+
+    const auto layout = gurka::detail::map_to_coordinates(ascii_map, gridsize);
+    speed_map = gurka::buildtiles(layout, ways, {}, {}, "test/data/fixedspeed");
+  }
+
+  inline float getDuration(const valhalla::Api& route) {
+    return route.directions().routes(0).legs(0).summary().time();
+  }
+
+  void doTests(const std::string& costing,
+               const std::unordered_map<std::string, std::string>& options) {
+
+    valhalla::Api default_route =
+        gurka::do_action(valhalla::Options::route, speed_map, {"A", "D"}, costing);
+    float default_time = getDuration(default_route);
+
+    valhalla::Api capped_route =
+        gurka::do_action(valhalla::Options::route, speed_map, {"A", "D"}, costing, options);
+    float capped_time = getDuration(capped_route);
+
+    gurka::assert::raw::expect_path(default_route, {"ABCD"});
+    gurka::assert::raw::expect_path(capped_route, {"AEFD"});
+    gurka::assert::raw::expect_eta(capped_route, 7.92f, 0.001f);
+    ASSERT_LT(capped_time, default_time);
+  }
+};
+
+gurka::map FixedSpeedTest::speed_map = {};
+
+TEST_F(FixedSpeedTest, AutoFixedSpeed) {
+  doTests("auto", {{"/costing_options/auto/fixed_speed", "50"}});
+}
+
+TEST_F(FixedSpeedTest, TruckFixedSpeed) {
+  doTests("truck", {{"/costing_options/truck/fixed_speed", "50"}});
+}
+
+TEST_F(FixedSpeedTest, MotorcycleFixedSpeed) {
+  doTests("motorcycle", {{"/costing_options/motorcycle/fixed_speed", "50"}});
+}
```

Comment:
would you mind adding a test covering the scenario where both `top_speed` and `fixed_speed` are specified?

Reply:
Sure thing, just added that test in the latest commit along with your other suggestions thanks.